File: UI/grooveon_desktop/lib/models/response/song_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'song_response.g.dart';

@JsonSerializable()
class SongResponse {
  final int id;
  final String? externalTrackId;
  final String source;
  final String title;

  final int artistId;
  final String artistName;

  final int? albumId;
  final String? albumTitle;

  final int? genreId;
  final String? genreName;

  final int durationSeconds;
  final String? previewUrl;
  final String? coverUrl;
  final DateTime? releaseDate;

  final bool isActive;
  final DateTime? lastSyncedAt;
  final DateTime createdAt;

  SongResponse({
    required this.id,
    this.externalTrackId,
    required this.source,
    required this.title,
    required this.artistId,
    required this.artistName,
    this.albumId,
    this.albumTitle,
    this.genreId,
    this.genreName,
    required this.durationSeconds,
    this.previewUrl,
    this.coverUrl,
    this.releaseDate,
    required this.isActive,
    this.lastSyncedAt,
    required this.createdAt,
  });

  factory SongResponse.fromJson(Map<String, dynamic> json) =>
      _$SongResponseFromJson(json);

  Map<String, dynamic> toJson() => _$SongResponseToJson(this);
}

File: GrooveOn.Models/ResponseObjects/SongResponse.cs
using System;

namespace GrooveOn.Model.Responses
{
    public class SongResponse
    {
        public int Id { get; set; }
        public string? ExternalTrackId { get; set; }
        public string Source { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;

        public int ArtistId { get; set; }
        public string ArtistName { get; set; } = string.Empty;

        public int? AlbumId { get; set; }
        public string? AlbumTitle { get; set; }

        public int DurationSeconds { get; set; }
        public string? PreviewUrl { get; set; }
        public string? CoverUrl { get; set; }
        public DateTime? ReleaseDate { get; set; }

        public bool IsActive { get; set; }
        public DateTime? LastSyncedAt { get; set; }
        public DateTime CreatedAt { get; set; }

        public string? WhyRecommended { get; set; }
    }
}